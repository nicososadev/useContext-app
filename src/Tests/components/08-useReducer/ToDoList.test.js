import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { ToDoList } from '../../../components/08-useReducer/ToDoList'
import { demoToDos } from '../../fixtures/demoToDos'

describe('Pruebas en el componente ToDoList', () => {
    
    const handleComplete = jest.fn()
    const handleDelete = jest.fn()

    const wrapper = shallow( 
        <ToDoList 
            toDos={ demoToDos }
            handleComplete={ handleComplete }
            handleDelete={ handleDelete }
        /> 
    )

    test('Debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe mostrar dos ToDoListItems', () => {
        
        expect( wrapper.find('ToDoListItem').length ).toBe( demoToDos.length )
        expect( wrapper.find('ToDoListItem').at(0).prop('handleDelete') ).toEqual( expect.any(Function) )
    })
    
})
