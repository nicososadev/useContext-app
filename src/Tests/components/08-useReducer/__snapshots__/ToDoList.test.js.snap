// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ToDoList Debe mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <ToDoListItem
    handleComplete={[MockFunction]}
    handleDelete={[MockFunction]}
    index={0}
    key="1"
    toDo={
      Object {
        "desc": "Prueba",
        "done": false,
        "id": 1,
      }
    }
  />
  <ToDoListItem
    handleComplete={[MockFunction]}
    handleDelete={[MockFunction]}
    index={1}
    key="2"
    toDo={
      Object {
        "desc": "Prueba2",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
